pkg/common/cns-lib/volume/manager.go

```diff
@@ -226,13 +234,15 @@ func (m *defaultManager) CreateVolume(ctx context.Context, spec *cnstypes.CnsVol
 		log.Errorf("taskResult is empty for CreateVolume task: %q", taskInfo.ActivationId)
 		return nil, errors.New("taskResult is empty")
 	}
+
 	volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 	if volumeOperationRes.Fault != nil {
 		fault, ok := volumeOperationRes.Fault.Fault.(cnstypes.CnsAlreadyRegisteredFault)
 		if ok {
 			log.Infof("CreateVolume: Volume is already registered with CNS. VolumeName: %q, volumeID: %q, opId: %q", spec.Name, fault.VolumeId.Id, taskInfo.ActivationId)
-			return &cnstypes.CnsVolumeId{
-				Id: fault.VolumeId.Id,
+			return &CnsVolumeInfo{
+				DatastoreURL: "",
```

Comment:
Why is this set to empty string? How would the client know when the DatastoreURL will be set and when it wont be?

Reply:
I think client can check whether datastoreURL is an empty string or not. Valid datastoreURL cannot be empty string.

Reply:
It looks like only in case of static provisioning or registering an existing disk, the `CreateVolume` method does not set the DatastoreURL in the `CnsVolumeInfo`. Can you add this as a doc to this method so that the callers know about this behavior?